tools/tools.go

```diff
@@ -0,0 +1,29 @@
+//go:build tools
+// +build tools
+
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file will never be built, but `go mod tidy` will see the packages
+// imported here as dependencies and not remove them from `go.mod`.
+
+package main
+
+import (
+	// krane is used in e2e tests to package and push OCI images.
+	// Since we are only using krane for testing, we don't need to fork it
+	// and set up an independent build pipeline, like we do for helm/kustomize.
+	// Instead, we can just vendor the source and build it locally.
```

Comment:
Do we need to vendor this? Why not just `go install github.com/google/go-containerregistry/cmd/krane`?

Reply:
I thought it might be a good way to keep krane up to date. This way it will be covered by the CVE warnings, which will cause it to be updated more frequently, even tho we don't ship it for production use. It also means the version is only in one place, which makes it easy to find and update. Plus we already vendor the parent project, so this keeps the versions in sync. 

But if you think it's too confusion or complex, we don't need to vendor it.

Reply:
I would prefer to install a specific version of the binary,  rather than to vendor it as a dependency. `krane` is one option to push OCI images, and we don't want to bundle it with the production code.